src/overlays/actors/ovl_En_Ma1/z_en_ma1.c

```diff
@@ -19,45 +36,412 @@ const ActorInit En_Ma1_InitVars = {
+        if (this->skelAnime.animCurrentSeg != &D_06000820) {
+            func_80AA0A84(this, 1);
+        }
+    } else {
+        if (this->skelAnime.animCurrentSeg != &D_06008D64) {
+            func_80AA0A84(this, 3);
+        }
+    }
+
+    if ((globalCtx->sceneNum == 0x5F) && (gSaveContext.eventChkInf[1] & 0x10)) {
+        Actor_Kill(&this->actor);
+    } else if ((!(gSaveContext.eventChkInf[1] & 0x10)) || (CHECK_QUEST_ITEM(QUEST_SONG_EPONA))) {
+        if (this->unk_1E8.unk_00 == 2) {
+            this->actionFunc = (ActorFunc)func_80AA0EA0;
+            globalCtx->msgCtx.unk_E3E7 = 4;
+            globalCtx->msgCtx.msgMode = 0x36;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ma1/func_80AA08C4.s")
+void func_80AA0EA0(EnMa1* this, GlobalContext* globalCtx) {
+    if (func_8002F410(&this->actor, globalCtx)) {
+        this->actor.attachedA = NULL;
+        this->actionFunc = (ActorFunc)func_80AA0EFC;
+    } else {
+        func_8002F434(&this->actor, globalCtx, GI_WEIRD_EGG, 120.0f, 10.0f);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ma1/func_80AA0A0C.s")
+void func_80AA0EFC(EnMa1* this, GlobalContext* globalCtx) {
+    if (this->unk_1E8.unk_00 == 3) {
+        this->unk_1E8.unk_00 = 0;
+        this->actionFunc = (ActorFunc)func_80AA0D88;
+        gSaveContext.eventChkInf[1] |= 4;
+        globalCtx->msgCtx.msgMode = 0x36;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ma1/func_80AA0A84.s")
+void func_80AA0F44(EnMa1* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ma1/func_80AA0AF4.s")
+    if (this->unk_1E8.unk_00 != 0) {
+        if (this->skelAnime.animCurrentSeg != &D_06000820) {
+            func_80AA0A84(this, 1);
+        }
+    } else {
+        if (this->skelAnime.animCurrentSeg != &D_06008D64) {
+            func_80AA0A84(this, 3);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ma1/func_80AA0B74.s")
+    if (gSaveContext.eventChkInf[1] & 0x40) {
+        if (player->stateFlags2 & 0x1000000) {
+            player->stateFlags2 |= 0x2000000;
+            player->unk_6A8 = &this->actor;
+            this->actor.textId = 0x2061;
+            func_8010B680(globalCtx, this->actor.textId, 0);
+            this->unk_1E8.unk_00 = 1;
+            this->actor.flags |= 0x10000;
+            this->actionFunc = (ActorFunc)func_80AA106C;
+        } else if (this->actor.xzDistanceFromLink < 30.0f + (f32)this->collider.dim.radius) {
+            player->stateFlags2 |= 0x800000;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ma1/EnMa1_Init.s")
+void func_80AA106C(EnMa1* this, GlobalContext* globalCtx) {
+    PLAYER->stateFlags2 |= 0x800000;
+    if (this->unk_1E8.unk_00 == 2) {
+        func_800ED858(2);
+        func_8010BD58(globalCtx, 9);
+        this->actor.flags &= 0xFFFEFFFF;
```

Comment:
Perhaps use ~0x10000 instead? It's more representative of what this code is doing, since it is toggling that flag off.

Reply:
gotcha, yeah